Paranormal/Paranormal/Panels/ToolSettings/ActiveToolSettingsViewController.swift

```diff
@@ -52,27 +50,32 @@ public class ActiveToolSettingsViewController: PNViewController {
         }
     }
 
-    public var brushSize: Float = 30.0 {
-        didSet {
-            document?.brushSize = brushSize
-        }
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        updateSettingsSliders()
     }
 
-    public var brushOpacity: Float = 100.0 {
-        didSet(value) {
-            document?.brushOpacity = brushOpacity
-        }
+    @IBAction func updateSize(sender: NSSlider) {
+        document?.toolSettings.size = sizeSlider.floatValue
     }
 
-    public var brushHardness: Float = 0.9 {
-        didSet(value) {
-            document?.brushHardness = brushHardness
-        }
+    @IBAction func updateStrength(sender: NSSlider) {
+        document?.toolSettings.strength = strengthSlider.floatValue
+    }
+
+    @IBAction func updateHardness(sender: NSSlider) {
+        document?.toolSettings.hardness = hardnessSlider.floatValue
     }
 
-    public var gaussianRadius: Float = 10.0 {
-        didSet(value) {
-            document?.gaussianRadius = gaussianRadius
+    public func updateSettingsSliders() {
+        if let doc = document {
+            sizeSlider.floatValue = doc.toolSettings.size
+            strengthSlider.floatValue = doc.toolSettings.strength
+            hardnessSlider.floatValue = doc.toolSettings.hardness
+        }
+
+        if let cpController = colorPickerViewController? {
+            cpController.updateColorPicker()
```

Comment:
luke's I wish everything were functional nit: Not a huge deal, but it would be nice if this didn't rely on the globalish document state. maybe updateColorPicker(toolsettings).